src/test/groovy/pl/allegro/tech/opel/OpelEngineSpec.groovy

```diff
@@ -749,25 +749,150 @@ xyz'"""
         given:
         def parentContext = EvalContext.Builder.create()
                 .withCompletedVariable('myVar', 1)
-                .withFunction('myFunc', { apply << CompletableFuture.completedFuture(1) })
+                .withFunction('myFunc', constFunctionReturning(1))
                 .build()
 
         def evalContext = EvalContext.Builder.create()
                 .withParentEvalContext(parentContext)
                 .withCompletedVariable('myVar', 100)
-                .withFunction('myFunc', { apply << CompletableFuture.completedFuture(100) })
+                .withFunction('myFunc', constFunctionReturning(100))
                 .build();
 
         when:
-        def eval = new OpelEngine().eval('2+myVar', evalContext).get()
+        def result = new OpelEngine().eval('2+myVar', evalContext).get()
 
         then:
-        eval == 102
+        result == 102
 
         where:
         input << ['2+myVar', '2+myFunc']
     }
 
+    @Unroll
+    def "should evaluate expression with variable and function registered in engine"() {
```

Comment:
OpelEngineSpec is integration test so I suggest to:
- rename it to OpelEngineIntegrationSpec
- cut it to smaller specs, eg. one for arithmetic, other for function and variable registration